src/components/CytoscapeGraph/CytoscapeGraph.tsx

```diff
@@ -859,6 +861,98 @@ export class CytoscapeGraph extends React.Component<CytoscapeGraphProps, Cytosca
         console.error(`Could not fetch health for [${ele.data(CyNode.nodeType)}] [${key}]: ${API.getErrorString(err)}`);
       });
   }
+
+  private fixLoopOverlap(cy: any) {
+    cy.$(':loop').forEach(loop => {
+      const node = loop.source();
+      const otherEdges = node.connectedEdges().subtract(loop);
+      const minDistance = 1;
+
+      // Default values in rads (taken from cytoscape docs)
+      const DEFAULT_LOOP_SWEEP = -1.5707;
+      const DEFAULT_LOOP_DIRECTION = -0.7854;
+
+      loop.style('loop-direction', DEFAULT_LOOP_DIRECTION);
+      loop.style('loop-sweep', DEFAULT_LOOP_SWEEP);
+
+      let found = false;
+      // Check if we have any other edge that overlaps with any of our loop edges
+      otherEdges.forEach(edge => {
+        const testPoint = edge.source().same(node) ? edge.sourceEndpoint() : edge.targetEndpoint();
+        if (
+          squaredDistance(testPoint, loop.sourceEndpoint()) <= minDistance ||
+          squaredDistance(testPoint, loop.targetEndpoint()) <= minDistance
+        ) {
+          found = true;
+          return false; // break the cytoscape forEach
+        }
+        return true;
```

Comment:
Is this line necessary?

Reply:
I got a "not all code paths return a value" error, but could update the function to signal that it returns false or undefined

Reply:
ok